test/parallel/test-process-emitwarning.js

```diff
@@ -24,6 +26,10 @@ function CustomWarning() {
 util.inherits(CustomWarning, Error);
 process.emitWarning(new CustomWarning());
 
+const warningNoToString = new CustomWarning();
+warningNoToString.toString = null;
```

Comment:
I wonder if we should also handle cases where `toString()` is a function but throws?